verl/workers/rollout/sglang_rollout/async_sglang_server.py

```diff
@@ -36,37 +36,42 @@ def __init__(self, config: DictConfig, dp_size: int, dp_rank: int, wg_prefix: st
         self._dp_rank = dp_rank
         self.wg_prefix = wg_prefix
         self.workers = []
+        self.master_worker = None
 
     async def init_engine(self):
+        if self.workers:
+            # avoid init twice
+            return
         all_actors = ray.util.list_named_actors(all_namespaces=True)
         matched_actors = [actor for actor in all_actors if actor.get("name", None).startswith(self.wg_prefix + "WorkerDict_")]
 
-        # TODO support multi node
         for matched_actor in matched_actors:
-            current_rank = int(matched_actor["name"].split(":")[-1])
+            fields = matched_actor["name"].split(":")
+            assert len(fields) == 2, f"invalid actor name: {matched_actor['name']}"
+            pg_index, local_rank = int(fields[0].split("_")[-1]), int(fields[1])
```

Comment:
Does the current implementation support cross-node inference?

Reply:
yes, I have tested this case.